xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_crypto.h

```diff
@@ -116,14 +116,8 @@ extern "C"
     // Flags to use special conditions, see STREAM_CRYPTO_FLAGS for available flags.
     uint8_t flags;
 
-    // The size of the crypto session key id set in sessionId.
-    uint16_t sessionIdSize;
-
     // The crypto session key id.
-    //
-    // WARNING The data set here is not copied and its pointer must still be
-    // available.
-    const char* sessionId;
+    char sessionId[256];
```

Comment:
As question, is it never longer as 256 bytes?
Not know exact this crypto session system :sweat_smile:, if them never bigger this is really the best, thanks. Becomes always hard to pass a string pointer from outside and to make sure not changed or deleted.

Reply:
The ids I've seen were all 32 chars long. No idea how long they actually can get. Maybe @peak3d knows?

Reply:
yes, 32 byte / chars